tests/test_ui_mpldrawer.py

```diff
@@ -0,0 +1,342 @@
+
+
+def test_plot_circuit_error_style():
+    """Test for style error function"""
+    style1 = _plot_params(style="test")
+    style2 = _plot_params(style="fardelejo")
+    custom_style = {
+        "facecolor": "#6497bf",
+        "edgecolor": "#01016f",
+        "linecolor": "#01016f",
+        "textcolor": "#01016f",
+        "fillcolor": "#ffb9b9",
+        "gatecolor": "#d8031c",
+        "controlcolor": "#360000",
+    }
+    style3 = _plot_params(style=custom_style)
+    assert style1["facecolor"] == "w"
+    assert style2["facecolor"] == "#e17a02"
+    assert style3["facecolor"] == "#6497bf"
+
+
+@pytest.mark.parametrize("qubits", [3, 4, 5, 6])
+def test_bigger_circuit_gates(qubits):
+    """Test for a bigger circuit"""
+    c = Circuit(qubits)
+    c.add(gates.H(1))
+    c.add(gates.X(1))
+    c.add(gates.SX(2))
+    c.add(gates.CSX(0, 2))
+    c.add(gates.TOFFOLI(0, 1, 2))
+    c.add(gates.CNOT(1, 2))
+    c.add(gates.SWAP(1, 2))
+    c.add(gates.SiSWAP(1, 2))
+    c.add(gates.FSWAP(1, 2))
+    c.add(gates.DEUTSCH(1, 0, 2, np.pi))
+    c.add(gates.X(1))
+    c.add(gates.X(0))
+    c.add(gates.M(qubit) for qubit in range(2))
+    ax, _ = plot_circuit(c)
+    assert ax != None
+
+
+@pytest.mark.parametrize("clustered", [False, True])
+def test_complex_circuit(clustered):
+    """Complex circuits for several cases"""
+    c = Circuit(3)
+    c.add(gates.H(0))
+    c.add(gates.H(1))
+    c.add(gates.H(2))
+    c.add(gates.X(1))
+    c.add(gates.Z(0))
+    c.add(gates.CNOT(0, 1))
+    c.add(gates.CZ(0, 1))
+    c.add(gates.CRX(0, 1, np.pi))
+    c.add(gates.Y(1))
+    c.add(gates.RY(1, np.pi))
+    c.add(gates.CRY(1, 2, np.pi))
+    c.add(gates.Z(1))
+    c.add(gates.SX(2))
+    c.add(gates.CSX(0, 2))
+    c.add(gates.X(0))
+    c.add(gates.TOFFOLI(0, 1, 2))
+    c.add(gates.X(0))
+    c.add(gates.CNOT(1, 2))
+    c.add(gates.SWAP(1, 2))
+    c.add(gates.SWAP(1, 2).dagger())
+    c.add(gates.SX(1).dagger())
+    c.add(gates.X(0))
+    c.add(gates.X(2))
+    c.add(gates.H(0))
+    c.add(gates.SiSWAP(1, 2).dagger())
+    c.add(gates.FSWAP(1, 2).dagger())
+    c.add(gates.DEUTSCH(1, 0, 2, np.pi))
+    c.add(gates.X(0))
+    c.add(gates.M(qubit) for qubit in range(2))
+    ax1, _ = plot_circuit(c.invert(), cluster_gates=clustered, scale=0.70)
+    ax2, _ = plot_circuit(c, cluster_gates=clustered, scale=0.70)
+    assert ax1 != None
+    assert ax2 != None
+
+
+def test_align_gate():
+    """Test for Align gate"""
+    c = Circuit(3)
+    c.add(gates.Align(0))
+    ax, _ = plot_circuit(c)
+    assert c != None
+    assert ax != None
+
+
+def test_fused_gates():
+    """Test for FusedStartGateBarrier and FusedEndGateBarrier"""
+    min_q = 0
+    max_q = 1
+    l_gates = 1
+    equal_qbits = True
+    start_barrier = FusedStartGateBarrier(min_q, max_q, l_gates, equal_qbits)
+    end_barrier = FusedEndGateBarrier(min_q, max_q)
+    assert start_barrier.unitary == start_barrier.unitary
+    assert end_barrier.unitary == end_barrier.unitary

```

Comment:
Was the idea here to check whether end and start barriers are the same? I wouldn't take as a valid check to use an assertion on the same object.

Reply:
The idea is to create a dummy check, I do not bare in mind to do something odd, but you are right, it could be also an assertion on the same object